### PR TITLE
Feat/admin account management

### LINKED ISSUE #26: Admin account management
- Admin is able to change his/her password
- Admin is able to delete other admin's account

### PR DESCRIPTION
**Pull request by WiktorSmura**
- Added new page with form to change password of currently logged in admin
- Modified `/admin/admins.tsx` page so now you can delete other admins

Closes #26

### CODE REVIEW

File: pages/api/admins.ts [hunk 1]
```diff
@@ -36,5 +40,20 @@ export default withApiAuth(
...
     }),
+    DELETE: withApiValidation({ body: deleteAdminSchema }, async (req, res) => {
+      const { ids } = req.body;
+
+      try {
+        ids.forEach(async (adminId) => {
+          await prisma.admin.delete({
+            where: { id: adminId },
+          });
+        });
```
Comment (AdamSiekierski) [inline pages/api/admins.ts:51]: This should be wrapped into one database transaction https://www.prisma.io/docs/concepts/components/prisma-client/transactions

```suggestion
        await prisma.$transaction(ids.map((adminId) => prisma.admin.delete({
            where: { id: adminId },
          });
        ));
```

File: pages/api/changePassword.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  newPassword: string().required("New password is required"),
+  hashedOldPassword: string().required(),
+  id: string().required("User id is required"),
+});
+
+export type ChangePasswordType = Asserts<typeof changePasswordSchema>;
+
+export default withApiAuth(
+  withApiMethods({
+    POST: withApiValidation({ body: changePasswordSchema }, async (req, res) => {
```
Comment (datejer) [inline pages/api/changePassword.ts:19]: Maybe we should use the `PATCH` method here?

File: pages/api/changePassword.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+          return res.json({
+            message: "Admin password has been changed successfully",
+            changedAdmin,
+          });
+        } catch {
+          return res.status(500).json({ message: "Internal server error" });
+        }
+      } else {
+        return res.json({ errorMessage: "Bad old password" });
```
Comment (AdamSiekierski) [inline pages/api/changePassword.ts:39]: Let's stick to the methodology we went with earlier - the error messages are sent in the `message` field of the response.

Also, set the response status to an error one - I'd suggest `422`

File: pages/api/changePassword.ts (hunk 2 above)
Comment (WiktorSmura) [reply pages/api/changePassword.ts:19]: I was talking about it with @AdamSiekierski, and he said we will better do it with POST method :smile:

File: pages/api/changePassword.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+});
+
+export type ChangePasswordType = Asserts<typeof changePasswordSchema>;
+
+export default withApiAuth(
+  withApiMethods({
+    POST: withApiValidation({ body: changePasswordSchema }, async (req, res) => {
+      const { oldPassword, newPassword, hashedOldPassword, id } = req.body;
+
+      if (await compare(oldPassword, hashedOldPassword)) {
```
Comment (AdamSiekierski) [inline pages/api/changePassword.ts:22]: Wrapping the `compare` in a try ... catch statement may be a good idea

**Review by AdamSiekierski — CHANGES_REQUESTED** [review]
Well done! 🔥 Just a few minor things to fix, but seeing you do so much progress in a so short time is amazing!

File: pages/api/changePassword.ts (hunk 2 above)
Comment (datejer) [reply pages/api/changePassword.ts:19]: Ah, okay then 👍

Comment (AdamSiekierski) [reply pages/api/changePassword.ts:19]: POST is an appropriate method here. Only PUT may be discussable, but I'd stick to POST in this case (this endpoint isn't RESTful anyways, and it doesn't really make sense to make it so)

**Review by AdamSiekierski — APPROVED** [review]
LGTM 🚀

**Pull request merged**